### PR TITLE
[StyleCop] Fix all warnings in the NumberWithUnit basic files

### PR DESCRIPTION
**Pull request by Aliandi**
- Supress warning on variables name format to be able to use underscore in the constant names
- Remove whitespaces
- Document enums
- Rename field to begin with uppercase after changing it to propety
- Add class access modifier
- Use string.Empty for empty strings

### CODE REVIEW

**Review by ceciliaavila — CHANGES_REQUESTED** [review]
Please, check the last commit message. It says the opposite of what was done.

**Comment by Aliandi** [comment]
@ceciliaavila Commit message changed!

**Review by ceciliaavila — APPROVED** [review]
LGTM

File: .NET/Microsoft.Recognizers.Text.NumberWithUnit/Parsers/CurrencyUnitValue.cs [hunk 1]
```diff
@@ -1,7 +1,19 @@
...
+﻿using System.Collections.Generic;
+
+using Microsoft.Recognizers.Text.NumberWithUnit.Utilities;
+
+namespace Microsoft.Recognizers.Text.NumberWithUnit
 {
-    public class CurrencyUnitValue : UnitValue
+    public class CurrencyUnitValue
     {
-        public string IsoCurrency { get; set; } = string.Empty;
```
Comment (VictorGrycuk) [inline .NET/Microsoft.Recognizers.Text.NumberWithUnit/Parsers/CurrencyUnitValue.cs]: Why was the auto property removed?
Th following format does not raise an warning on my side:
```
public string Number { get; set; } = string.Empty;

public string Unit { get; set; } = string.Empty;

public string IsoCurrency { get; set; } = string.Empty;
```

**Review by VictorGrycuk — APPROVED** [review]
LGTM.
However left a comment asking for clarification.

File: .NET/Microsoft.Recognizers.Text.NumberWithUnit/Parsers/CurrencyUnitValue.cs [hunk 2]
```diff
@@ -1,7 +1,19 @@
-﻿namespace Microsoft.Recognizers.Text.NumberWithUnit
+﻿using System.Collections.Generic;
+
+using Microsoft.Recognizers.Text.NumberWithUnit.Utilities;
+
+namespace Microsoft.Recognizers.Text.NumberWithUnit
 {
-    public class CurrencyUnitValue : UnitValue
+    public class CurrencyUnitValue
```
Comment (JuanAr) [inline .NET/Microsoft.Recognizers.Text.NumberWithUnit/Parsers/CurrencyUnitValue.cs]: This class inherited from UnitValue. Why it does not anymore?

**Review by JuanAr — CHANGES_REQUESTED** [review]
There seems to be a public class hierarchy change.

File: .NET/Microsoft.Recognizers.Text.NumberWithUnit/Parsers/CurrencyUnitValue.cs (hunk 2 above)
Comment (Aliandi) [reply .NET/Microsoft.Recognizers.Text.NumberWithUnit/Parsers/CurrencyUnitValue.cs]: It was a bad merge, fixed

File: .NET/Microsoft.Recognizers.Text.NumberWithUnit/Parsers/CurrencyUnitValue.cs (hunk 1 above)
Comment (Aliandi) [reply .NET/Microsoft.Recognizers.Text.NumberWithUnit/Parsers/CurrencyUnitValue.cs]: It was a bad merge, fixed

File: .NET/Microsoft.Recognizers.Text.NumberWithUnit/Parsers/NumberWithUnitParser.cs [hunk 3]
```diff
@@ -8,10 +8,10 @@ public class NumberWithUnitParser : IParser
     {
         public NumberWithUnitParser(INumberWithUnitParserConfiguration config)
         {
-            this.config = config;
+            this.Config = config;
         }
 
-        protected INumberWithUnitParserConfiguration config { get; private set; }
```
Comment (JuanAr) [inline .NET/Microsoft.Recognizers.Text.NumberWithUnit/Parsers/NumberWithUnitParser.cs]: I think we should extract this change into the Breaking changes PR we created for this type of warnings. The thing is we are renaming a protected property.

**Review by JuanAr — CHANGES_REQUESTED** [review]
Some feedback regarding the renaming of a protected property. I thing we already started consolidating these changes into a single PR.

**Comment by JuanAr** [comment]
@Aliandi sorry, another feedback item I didn't see before.

File: .NET/Microsoft.Recognizers.Text.NumberWithUnit/Parsers/NumberWithUnitParser.cs (hunk 3 above)
Comment (Aliandi) [reply .NET/Microsoft.Recognizers.Text.NumberWithUnit/Parsers/NumberWithUnitParser.cs]: Feedback 🆗

**Review by JuanAr — APPROVED** [review]
👍

**Pull request closed without merging**